lib/rules/no-undef.js

```diff
@@ -90,6 +90,9 @@ module.exports = function(context) {
             scope = scope.upper;
             variables = scope.variables.concat(variables);
         }
+        if (scope.childScopes.length) {
```

Comment:
Same comment as in the other PR - we can't just do this for every scope. It must be limited to the global scope.